# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/xre/nsAppRunner.cpp
# Commit: 0895df8f4a77
# Full Hash: 0895df8f4a7708200f6b752e798c609628b39650
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-04-30 21:45:04
# Regressor Bug: 1707655
# File Overlap Count: 1
# Description:
#   Bug 1707655 - Move the activation of supplemental fonts on macOS to a separate thread and try to get it started earlier. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113552
# ==============================================================================

diff -r 097daf3a851d -r 0895df8f4a77 toolkit/xre/nsAppRunner.cpp
--- a/toolkit/xre/nsAppRunner.cpp	Fri Apr 30 13:11:37 2021 +0000
+++ b/toolkit/xre/nsAppRunner.cpp	Fri Apr 30 13:50:47 2021 +0000
@@ -92,6 +92,9 @@
 #include "mozilla/dom/quota/QuotaManager.h"
 #include "mozilla/scache/StartupCache.h"
 #include "gfxPlatform.h"
+#ifdef XP_MACOSX
+#  include "gfxPlatformMac.h"
+#endif
 
 #include "mozilla/Unused.h"
 
@@ -5417,6 +5420,13 @@
   AUTO_PROFILER_INIT;
   AUTO_PROFILER_LABEL("XREMain::XRE_main", OTHER);
 
+#ifdef XP_MACOSX
+  // We call this early because it will kick off a background-thread task
+  // to register the fonts, and we'd like it to have a chance to complete
+  // before gfxPlatform initialization actually requires it.
+  gfxPlatformMac::RegisterSupplementalFonts();
+#endif
+
 #ifdef MOZ_CODE_COVERAGE
   CodeCoverageHandler::Init();
 #endif